core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/UnwrapYearInComparison.java

```diff
@@ -124,6 +129,36 @@ public Expression rewriteComparisonExpression(ComparisonExpression node, Void co
             return unwrapYear(expression);
         }
 
+        @Override
+        public Expression rewriteInPredicate(InPredicate node, Void context, ExpressionTreeRewriter<Void> treeRewriter)
+        {
+            Expression inPredicate = treeRewriter.defaultRewrite((Expression) node, null);
+            Expression value = node.getValue();
+            Expression valueList = node.getValueList();
```

Comment:
```suggestion
            Expression value = inPredicate.getValue();
            Expression valueList = inPredicate.getValueList();
```

Reply:
done. I wasn't sure that the returned expression from `defaultRewriter` must be an `InPredicate` so I also added a validation.